packages/realm/src/app-services/BaseSubscriptionSet.ts

```diff
@@ -0,0 +1,216 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { MutableSubscriptionSet, Realm, Subscription, SubscriptionSet, assert, binding } from "../internal";
+
+/**
+ * Enum representing the state of a {@link SubscriptionSet}.
+ */
+export enum SubscriptionsState {
```

Comment:
Cannot currently make this `const` due to how it is used in the `Sync` namespace in `Sync.ts`.

Reply:
I think I remember having discussions about this naming.  I do not remember where I landed, but I feel `SubscriptionSet` is more readable than `Subscriptions`.

Reply:
@elle-j 

> renaming of this type is possible for v12

Please bring it up at our next BindGen meeting

Reply:
@RedBeard0531 I mean `SubscriptionStateSet`